aesara/tensor/random/op.py

```diff
@@ -399,3 +399,43 @@ def grad(self, inputs, outputs):
 
     def R_op(self, inputs, eval_points):
         return [None for i in eval_points]
+
+
+class RandomMakerOp(Op):
+    def make_node(self, seed=None):
+        # np.inf is used as an representative of seed = None
+        if seed is None:
+            seed = np.inf
+        seed = as_tensor_variable(seed)
+        inputs = [seed]
+        outputs = [self.random_type()]
+        return Apply(self, inputs, outputs)
+
+    def perform(self, node, inputs, output_storage):
+        (seed,) = inputs
+        # np.inf is used as an representative of seed = None
+        if seed == np.inf:
+            seed = None
+        output_storage[0][0] = self.bitgen(seed=seed)
```

Comment:
This implies that there should be an abstract method named `bitgen` in the interface defined by `RandomMakerOp`.  It can be added using [`abc.abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod).

However, it doesn't look like an abstract implementation of `Op.perform` is necessary.  When `np.inf` is replaced by `None` (see above), there's no code duplicated between subclasses, because only a direct call the underlying RNG constructor is required.

Reply:
I made `bitgen` rapper function around those constructors primarily because directly having those constructors as class methods passed `self` into them by default. But I think the current approach works as well.